java/src/jmri/jmrit/display/Editor.java

```diff
@@ -2693,9 +2693,8 @@ public void dispose() {
      */
     public void showToolTip(Positionable selection, MouseEvent event) {
         ToolTip tip = selection.getToolTip();
-        String txt = tip.getText();
-        if (txt == null) {
-            tip.setText(selection.getNameString());
+        if (tip.getText() == null || tip.getText().isEmpty()) {
```

Comment:
This will eventually get flagged by SpotBugs. In a multi-threaded app, `tip.getText()` is not guaranteed to be the same in every call (even in a single check like this). That's why prior statement retained the result of `getText()` in a String and did comparisons on that. You may want to revert to doing that.

Reply:
OK, thanks. I'll do them both that way.